editor/plugins/theme_editor_plugin.cpp

```diff
@@ -31,12 +31,696 @@
+			edit_type_list->select(e_idx);
+			item_reselected = true;
+		}
+		e_idx++;
+	}
+	if (!item_reselected) {
+		edited_item_type = "";
+
+		if (edit_type_list->get_item_count() > 0) {
+			edit_type_list->select(0);
+		}
+	}
+
+	List<StringName> default_types;
+	base_theme->get_type_list(&default_types);
+	default_types.sort_custom<StringName::AlphCompare>();
+
+	edit_add_class_options->clear();
+	for (List<StringName>::Element *E = default_types.front(); E; E = E->next()) {
+		edit_add_class_options->add_item(E->get());
+	}
+
+	String selected_type = "";
+	Vector<int> selected_ids = edit_type_list->get_selected_items();
+	if (selected_ids.size() > 0) {
+		selected_type = edit_type_list->get_item_text(selected_ids[0]);
+
+		edit_items_add_color->set_disabled(false);
+		edit_items_add_constant->set_disabled(false);
+		edit_items_add_font->set_disabled(false);
+		edit_items_add_font_size->set_disabled(false);
+		edit_items_add_icon->set_disabled(false);
+		edit_items_add_stylebox->set_disabled(false);
+
+		edit_items_remove_class->set_disabled(false);
+		edit_items_remove_custom->set_disabled(false);
+		edit_items_remove_all->set_disabled(false);
+	} else {
+		edit_items_add_color->set_disabled(true);
+		edit_items_add_constant->set_disabled(true);
+		edit_items_add_font->set_disabled(true);
+		edit_items_add_font_size->set_disabled(true);
+		edit_items_add_icon->set_disabled(true);
+		edit_items_add_stylebox->set_disabled(true);
+
+		edit_items_remove_class->set_disabled(true);
+		edit_items_remove_custom->set_disabled(true);
+		edit_items_remove_all->set_disabled(true);
+	}
+	_update_edit_item_tree(selected_type);
+}
+
+void ThemeItemEditorDialog::_edited_type_selected(int p_item_idx) {
+	String selected_type = edit_type_list->get_item_text(p_item_idx);
+	_update_edit_item_tree(selected_type);
+}
+
+void ThemeItemEditorDialog::_update_edit_item_tree(String p_item_type) {
+	edited_item_type = p_item_type;
+
+	edit_items_tree->clear();
+	TreeItem *root = edit_items_tree->create_item();
+
+	List<StringName> names;
+
+	{
+		names.clear();
+		edited_theme->get_color_list(p_item_type, &names);
+
+		if (names.size() > 0) {
+			TreeItem *color_root = edit_items_tree->create_item(root);
+			color_root->set_metadata(0, Theme::DATA_TYPE_COLOR);
+			color_root->set_icon(0, get_theme_icon("Color", "EditorIcons"));
+			color_root->set_text(0, TTR("Colors"));
+			color_root->add_button(0, get_theme_icon("Clear", "EditorIcons"), ITEMS_TREE_REMOVE_DATA_TYPE, false, TTR("Remove All Color Items"));
+
+			names.sort_custom<StringName::AlphCompare>();
+			for (List<StringName>::Element *E = names.front(); E; E = E->next()) {
+				TreeItem *item = edit_items_tree->create_item(color_root);
+				item->set_text(0, E->get());
+				item->add_button(0, get_theme_icon("Edit", "EditorIcons"), ITEMS_TREE_RENAME_ITEM, false, TTR("Rename Item"));
+				item->add_button(0, get_theme_icon("Remove", "EditorIcons"), ITEMS_TREE_REMOVE_ITEM, false, TTR("Remove Item"));
+			}
+		}
+	}
```

Comment:
Any particular reason why this (and subsequent) code is in a sub-scope block? It doesn't have any local variable and blank line is enough for separating.

Reply:
If not for any other reason, I've likely put them in blocks so that they can be collapsed individually in the editor. If you don't think they should be there I can remove them.

Reply:
I'm not convinced that collapsing them is very useful. Although there's also no technical reason against doing that 🤔
It just looks weird to me, but style guide doesn't clarify this, so not sure.

Reply:
@KoBeWi IIRC these blocks are pretty much sprinkled a bit everywhere in the source, so I guess that's not an issue.

Reply:
I'm fine either way, it's nice to be able to visually identify sub-sections of the big method.

Either way, adding a comment for each to state what it's about (Colors, Fonts, etc.) would be good.

Reply:
To resolve this, I'll add comments to sections in #46808.